### PR TITLE
Update/0.10.1

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update to 0.10.1 (py3.11 compat)

https://github.com/justindujardin/pathy/tree/v0.10.1
https://github.com/justindujardin/pathy/blob/v0.10.1/requirements.txt

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ package:
 
 source:
   - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-    sha256: 838624441f799a06b446a657e4ecc9ebc3fdd05234397e044a7c87e8f6e76b1c
+    sha256: 4cd6e71b4cd5ff875cfbb949ad9fa5519d8d1dbe69d5fc1d1b23aa3cb049618b
 
 
 build:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:14]: `or win32` can be removed from skipping

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ requirements:
     - setuptools
   run:
     - python
-    - smart_open >=5.0.0,<6.0.0
+    - smart_open >=5.2.1,<6.0.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Should be `smart-open >=5.2.1,<7.0.0`, see https://github.com/justindujardin/pathy/blob/b39b49206518f7e51490d69a240236c40261aadd/requirements.txt#L1

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:14]: thank you, addressed.

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: thank you, addressed.

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**